segmentstore/storage/impl/src/main/java/io/pravega/segmentstore/storage/impl/filesystem/FileSystemStorage.java

```diff
@@ -409,7 +411,9 @@ private Throwable translateException(String segmentName, Throwable e) {
             retVal = new IllegalArgumentException(e.getMessage());
         }
 
-        if (e instanceof AccessDeniedException) {
+        if (e instanceof AccessControlException
+                || e instanceof AccessDeniedException
```

Comment:
In case of S3, we should have translated "AccessDenied" S3Exception. This is equivalent of `AccessDeniedException` for file system. Updated code to do this,